pkg/publicshare/manager/json/json.go

```diff
@@ -107,10 +107,10 @@ type manager struct {
 // CreatePublicShare adds a new entry to manager.shares
 func (m *manager) CreatePublicShare(ctx context.Context, u *user.User, rInfo *provider.ResourceInfo, g *link.Grant) (*link.PublicShare, error) {
 	id := &link.PublicShareId{
-		OpaqueId: randString(12),
+		OpaqueId: randString(15),
```

Comment:
is this by spec? Just curious

Reply:
I'm not sure if the spec tells about a minimum length. However that's the length we used in OC 10 and that the tests are enforcing. The longer the token, the safer it is as it will reduce collisions.

We might also need to look into "secure random" later on, as this is how tokens are generated in OC 10.